tests/test_coreg.py

```diff
@@ -366,3 +368,101 @@ def test_subsample(self):
         matrix_diff = np.abs(icp_full.to_matrix() - icp_sub.to_matrix())
         # Check that the x/y/z differences do not exceed 30cm
         assert np.count_nonzero(matrix_diff > 0.3) == 0
+
+    def test_apply_matrix(self):
+        warnings.simplefilter("error")
+        # This should maybe be its own function, but would just repeat the data loading procedure..
+
+        # Synthesize a shifted and vertically offset DEM
+        pixel_shift = 11
+        bias = 5
+        shifted_dem = self.ref.data.squeeze().copy()
+        shifted_dem[:, pixel_shift:] = shifted_dem[:, :-pixel_shift]
+        shifted_dem[:, :pixel_shift] = np.nan
+        shifted_dem += bias
+
+        matrix = np.diag(np.ones(4, dtype=float))
+        matrix[0, 3] = pixel_shift * self.tba.res[0]
+        matrix[2, 3] = -bias
+
+        transformed_dem = coreg.apply_matrix(shifted_dem.data.squeeze(), self.ref.transform, matrix)
+
+        diff = np.asarray(self.ref.data.squeeze() - transformed_dem)
+
+        # Check that the median is very close to zero
+        assert np.abs(np.nanmedian(diff)) < 0.05
```

Comment:
In a best case scenario, yes. But since cubic interpolation is done, there is actually a small difference. There could be a check in the start to see if the shift is by an integer amount (whereby it will only shift appropriately) or if the more complex warp should be made. Does that sound okay?

Reply:
Mmmh.. yes I guess it would be good to avoid resampling the DEM by too much when it's not necessary! Bilinear resampling has a good behaviour for this (if the shift is an integer, the resampling is right on the point, so same value). 
Ultimately this should be the choice of the user, so we can probably leave as is.

Reply:
Yes, if you think the issue comes from cubic interpolation, why not test it with bilinear interpolation here? I agree, somehow it should be exactly equal...

Reply:
Bilinear worked much better. I've set that as the default from now on in b1ffc81.